mirgecom/simutil.py

```diff
@@ -456,37 +781,279 @@ def generate_and_distribute_mesh(comm, generate_mesh):
     global_nelements : :class:`int`
         The number of elements in the serial mesh
     """
-    from meshmode.distributed import (
-        MPIMeshDistributor,
-        get_partition_by_pymetis,
-    )
-    num_parts = comm.Get_size()
-    mesh_dist = MPIMeshDistributor(comm)
-    global_nelements = 0
+    from warnings import warn
+    warn(
+        "generate_and_distribute_mesh is deprecated and will go away Q4 2022. "
+        "Use distribute_mesh instead.", DeprecationWarning, stacklevel=2)
+    return distribute_mesh(comm, generate_mesh, **kwargs)
+
+
+def distribute_mesh(comm, get_mesh_data, partition_generator_func=None):
+    r"""Distribute a mesh among all ranks in *comm*.
```

Comment:
This is included in #767. Do we want/need to add it here, or can it be omitted?

Reply:
It's a draft; I'd like to have a chance to go over it and fix things up before merging.

Reply:
No problem.   I'm just a little clunky at trying to remove it without removing it downstream.

Reply:
:tada: thanks matt